### PR TITLE
Fix S1192: Rule should display string as defined in the source code

### LINKED ISSUE #2191: Fix S1192: Rule should display string as defined in the source code
### Description

Analyzing Lucene.Net fails when reporting the issue about duplicated string for the char \u0000 on file [HTMLStripCharFilter.cs](https://raw.githubusercontent.com/apache/lucenenet/a3a12967b250e8e7e5f623f0ba7572ec64f479ac/src/Lucene.Net.Analysis.Common/Analysis/CharFilter/HTMLStripCharFilter.cs)

See scanner end step logs https://cirrus-ci.com/task/5111004193095680

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #2191

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs [hunk 1]
```diff
@@ -74,21 +74,22 @@ private void ReportOnViolation(SyntaxNodeAnalysisContext context)
                 .Cast<LiteralExpressionSyntax>();
 
             // Collect duplications
-            var stringWithLiterals = new Dictionary<string, List<LiteralExpressionSyntax>>();
+            var stringWithLiterals = new Dictionary<string, List<Location>>();
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs]: We only use the `Location` so there is no need to store the full object.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs [hunk 2]
```diff
@@ -74,21 +74,22 @@ private void ReportOnViolation(SyntaxNodeAnalysisContext context)
...
             foreach (var literal in stringLiterals)
             {
-                var stringValue = literal.Token.ValueText;
+                // Remove leading and trailing double quotes
+                var stringValue = literal.Token.Text.Substring(1, literal.Token.Text.Length - 2);
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs]: Actual change.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.cs [hunk 3]
```diff
@@ -4,6 +4,7 @@ namespace Tests.Diagnostics
 {
     public class Program
     {
+        private string empty = "";
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.cs]: This is to ensure that the substring doesn't fail for an empty string.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs (hunk 2 above)
Comment (valhristov) [reply sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs]: Does it work with strings like `@"asdasd"` or `$@"adasdasd"`? (not sure if those are different syntax kinds)

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs [hunk 4]
```diff
@@ -74,21 +74,22 @@ private void ReportOnViolation(SyntaxNodeAnalysisContext context)
...
                     !IsMatchingMethodParameterName(literal))
                 {
                     if (!stringWithLiterals.ContainsKey(stringValue))
                     {
-                        stringWithLiterals[stringValue] = new List<LiteralExpressionSyntax>();
+                        stringWithLiterals[stringValue] = new List<Location>();
                     }
 
-                    stringWithLiterals[stringValue].Add(literal);
+                    stringWithLiterals[stringValue].Add(literal.GetLocation());
```
Comment (valhristov) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs]: I think `GetLocation` is generally slow

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs (hunk 2 above)
Comment (Evangelink) [reply sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/StringLiteralShouldNotBeDuplicated.cs]: `$@""` and `$""` are not `StringLiteralExpression` and so not handled by this rule.

**Pull request merged**